tests/Kernel/Paragraphs/ParagraphsTest.php

```diff
@@ -465,4 +465,45 @@ public function quoteDataProvider(): array {
     ];
   }
 
+  /**
+   * Test 'contextual navigation' paragraph rendering.
+   */
+  public function testContextualNavigation(): void {
+    $paragraph = Paragraph::create([
+      'type' => 'oe_contextual_navigation',
+      'field_oe_title' => 'Contextual navigation',
+      'field_oe_links' => [
+        [
+          'title' => 'Link 1',
+          'uri' => 'internal:/',
+        ],
+        [
+          'title' => 'Link 2',
+          'uri' => 'internal:/',
+        ],
+      ],
+      'field_oe_limit' => 1,
+      'field_oe_text' => 'More links',
+    ]);
+
+    $paragraph->save();
+    $html = $this->renderParagraph($paragraph);
+
+    $crawler = new Crawler($html);
+    $actual = $crawler->filter('div.ecl-context-nav > span.ecl-context-nav__label')->text();
+    $this->assertEquals('Contextual navigation', trim($actual));
+
+    $actual = $crawler->filter('div.ecl-context-nav ul.ecl-context-nav__list a.ecl-link')->eq(0)->text();
+    $this->assertEquals('Link 1', trim($actual));
+
+    $actual = $crawler->filter('div.ecl-context-nav ul.ecl-context-nav__list a.ecl-link')->eq(1)->text();
+    $this->assertEquals('Link 2', trim($actual));
+
+    $actual = $crawler->filter('div.ecl-context-nav ul.ecl-context-nav__list li.ecl-context-nav__item--over-limit a.ecl-link')->eq(0)->text();
+    $this->assertEquals('Link 2', trim($actual));
+
+    $actual = $crawler->filter('div.ecl-context-nav ul.ecl-context-nav__list li.ecl-context-nav__more button')->text();
+    $this->assertEquals('More links', trim($actual));
```

Comment:
We don't test that links URLs are correctly printed out and, indeed, I don't see them in my manual tests:

![image](https://user-images.githubusercontent.com/153362/56492064-dc244480-64ea-11e9-84d5-0c183b9c65ef.png)

They all point to `#` while I set them to `/node` and `http://example.com`.